OpenVPN 2.4 update

angristan opened the pull request:
As OpenVPN 2.4 is out, some improvements will be implemented to the script.

- [x] Use AES-GCM for the data channel's cipher instead of AES-CBC
- [x] Disable DH
- [x] Enable ECDH
- [x] Use an ECDSA certificate instead of RSA
- [x] [Use --tls-crypt instead of --tls-auth](https://community.openvpn.net/openvpn/changeset/c6e24fa3e16c32f9b427e360fd07102f613aa5c6/)
- [x] ~~[Update the Arch Linux installation ](https://www.archlinux.org/news/openvpn-240-update-requires-administrative-interaction/)~~ Remove Arch Linux support
- [x] Not related to the 2.4 update, but I want to use the best encryption settings everywhere (AES 256 GCM and SHA 512), thus I'll remove the choice for other ciphers, thus making the script more simple to use.
- [ ] Update the Readme

Also, we have to make sure that OpenVPN 2.4 is available everywhere.
- [x] Debian 7 (thanks to the [official OpenVPN repo](https://community.openvpn.net/openvpn/wiki/OpenvpnSoftwareRepos))
- [x] Debian 8 (thanks to the [official OpenVPN repo](https://community.openvpn.net/openvpn/wiki/OpenvpnSoftwareRepos))
- [x] Ubuntu 12.04 LTS (thanks to the [official OpenVPN repo](https://community.openvpn.net/openvpn/wiki/OpenvpnSoftwareRepos))
- [x] Ubuntu 14.04 LTS (thanks to the [official OpenVPN repo](https://community.openvpn.net/openvpn/wiki/OpenvpnSoftwareRepos))
- [x] Ubuntu 16.04 LTS (thanks to the [official OpenVPN repo](https://community.openvpn.net/openvpn/wiki/OpenvpnSoftwareRepos))
- [ ] Ubuntu 16.10
- [ ] CentOS 6
- [ ] CentOS 7
- [x] Arch Linux

Clients : 
- [x] Windows
- [x] MacOS (via [Homebrew](https://github.com/Homebrew/homebrew-core/blob/master/Formula/openvpn.rb) or [TunnelBlick](https://tunnelblick.net/))
- [ ] iOS
- [x] Android with [an alternative client](https://play.google.com/store/apps/details?id=it.colucciweb.free.openvpn)
-  For Linux clients, see above.

angristan committed:
Update AES data channel ciphers

angristan committed:
Update tls-cipher to use ECDHE and ECDSA

angristan committed:
Disable DH and enable ECDH

cf. https://community.openvpn.net/openvpn/wiki/Openvpn24ManPage#lbAK

angristan committed:
Use secp256k1 curve for ECDH

cf. https://github.com/Angristan/OpenVPN-install/pull/33#issuecomment-280399196

angristan committed:
Update repo for Debian and Ubuntu

angristan committed:
Remove DH key size input, remove cipher input

And use AES-128-GCM as the data channel's cipher

angristan committed:
Remove cp for dh

angristan committed:
Use secp256k1 curve for ECDH (fix)

I forgot the curve in https://github.com/Angristan/OpenVPN-install/pull/33/commits/23daeca80a20ef2e237445351d57b83bfbc45c89

angristan committed:
Fix cipher

angristan committed:
Use ECDSA cert instead of RSA

angristan committed:
Replace tls-auth with tls-crypt

angristan committed:
Remove RSA key size input

angristan committed:
Remove Arch Linux support

Revert PR#2 :https://github.com/Angristan/OpenVPN-install/pull/2/files#diff-cda9722285f1718b319bb88e134e9efe

Too much work to do with OpenVPN 2.4 for a distro that is not much used on servers.

Kcchouette commented on openvpn-install.sh:
```diff
@@ -309,68 +222,41 @@ else
-				echo "#!/bin/bash" > $RCLOCAL
-			fi
-		fi
+	fi
```
I think you forgot to remove (see the next lines)
```
# Install dependencies
pacman -Syu openvpn iptables openssl wget ca-certificates curl --needed --noconfirm
systemctl enable iptables
systemctl start iptables
fi
fi
```

Kcchouette replied:
see #40

Kcchouette committed:
Remove the archlinux part

that @Angristan had forgotten

Kcchouette committed:
Remove the last $SYSCTL var

This var was indeed removed with the commit https://github.com/Angristan/OpenVPN-install/pull/33/commits/f5ec9f3a17e4fbe6017845668fad2f543f83ee06

angristan committed:
Merge pull request #40 from Kcchouette/patch-1

Remove the last archlinux part

angristan committed:
Use best encryption available

Does not affect speed significantly, if at all.

Kcchouette reviewed (approved):
.
